pipeline/backend/kubernetes/kubernetes.go

```diff
@@ -1,28 +1,44 @@
 package kubernetes
 
 import (
+	"bytes"
 	"context"
+	"fmt"
 	"io"
 	"os"
+	"strings"
+	"time"
 
 	"github.com/woodpecker-ci/woodpecker/pipeline/backend/types"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
+	// To authenticate to GCP K8s clusters
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 )
 
 type kube struct {
-	namespace string
-	endpoint  string
-	token     string
+	logs         *bytes.Buffer // TODO remove
```

Comment:
?

Reply:
I actually don't know how it should be resolved. I almost copied all of this code from laszlocphs PR #23.

Reply:
Ok lets add a **descriptive** TODO

Reply:
Okay I think I can just replace this "logs" buffer with zerolog trace log calls. It seems to be used for log messages like `Setting up Kubernetes primitives`, `Creating service`

Reply:
Thanks